src/contrib/dependencyinjection/Akka.DependencyInjection/ServiceProviderProps.cs

```diff
@@ -0,0 +1,63 @@
+// //-----------------------------------------------------------------------
+// // <copyright file="ServiceProviderProps.cs" company="Akka.NET Project">
+// //     Copyright (C) 2009-2021 Lightbend Inc. <http://www.lightbend.com>
+// //     Copyright (C) 2013-2021 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// // </copyright>
+// //-----------------------------------------------------------------------
+
+using System;
+using Akka.Actor;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Akka.DependencyInjection
+{
+    /// <summary>
+    /// This class represents a specialized <see cref="Akka.Actor.Props"/> that uses delegate invocation
+    /// to create new actor instances, rather than a traditional <see cref="System.Activator"/>.
+    ///
+    /// Relies on having an active <see cref="IServiceProvider"/> implementation available
+    /// </summary>
+    internal class ServiceProviderProps : Props
```

Comment:
We're going to remove this concept altogether in https://github.com/akkadotnet/akka.net/issues/4853 and replace it with an IIndirectActorProducer, similar to what we did in the original Akka.DI.Core library. 

We should probably merge that PR in first once it gets submitted, then rebase this on top of it.